Controls/Data.UWP/Data/Engine/Helpers/BindingExpressionHelper.cs

```diff
@@ -17,6 +18,53 @@ internal class BindingExpressionHelper
             return item => propertyInfo?.GetValue(item);
         }
 
+        public static Func<object, object> CreateGetNestedValueFunc(List<PropertyInfo> propertyInfos, string baseClassName)
+        {
+            return item =>
+            {
+                if (item == null)
+                {
+                    return null;
+                }
+
+                PropertyInfo basePropertyInfo = item.GetType().GetProperty(baseClassName);
+                item = basePropertyInfo.GetValue(item);
+
+                foreach (PropertyInfo propertyInfo in propertyInfos)
+                {
+                    item = propertyInfo.GetValue(item);
+                }
+
+                return item;
+            };
+        }
+
+        public static Action<object, object> CreateSetNestedValueFunc(List<PropertyInfo> propertyInfos, string baseClassName)
+        {
+            return new Action<object, object>((item, propertyValue) =>
+            {
+                if (item == null)
+                {
+                    return;
+                }
+
+                PropertyInfo basePropertyInfo = item.GetType().GetProperty(baseClassName);
+                item = basePropertyInfo.GetValue(item);
+
+                for (int i = 0; i < propertyInfos.Count - 1; i++)
+                {
+                    PropertyInfo info = propertyInfos[i];
+                    item = info.GetValue(item);
+                }
+
+                PropertyInfo propertyInfo = propertyInfos.Last();
+                if (propertyInfo.CanWrite)
```

Comment:
Can check CanWrite before even calling this function and skip it if not writable.